### PR TITLE
Replace string types with enums

### LINKED ISSUE #270: Replace string types with enum types.
VRR is using enums at some parts. Currently our types are using strings there as there seems to be problems mapping from one enum to another identical enum.

Probably the best/easiest solution is using one big switch statement to sort out which value of the VRR enum is mapped to which value of our custom enum (which does not necessarily need to be identical to the VRR one). 

Swagger/OpenAPI is supplying the field "enum" instead of "type" inside the @ApiProperty to define that the API is requesting/returning an enum value.

### PR DESCRIPTION
**Pull request by Paulomart**
Closes #270

### Description

This is the Pull Request Template for the Oeffis Project

## Checklist

- [ ] My code follows the style guidelines
- [ ] I have documented my code
- [ ] My changes generate no new warnings
- [ ] I have linked the Issue
- [ ] The Pull Request covers all the Checkpoints from the tagged Issue
- [ ] I have noted the hours of work i have put in the issue

## I have edited following Parts of the project

- [ ] Frontend
- [ ] Backend
- [ ] Github
- [ ] Root

## If tests are required did you make them?

- [ ] No tests required
- [ ] Yes
- [ ] No

### CODE REVIEW

File: backend/src/main.ts [hunk 1]
```diff
@@ -14,7 +14,7 @@ async function bootstrap(): Promise<void> {
     .build();
 
   const document = SwaggerModule.createDocument(app, config);
-  fs.writeFileSync("./swagger-spec.json", JSON.stringify(document));
+  fs.writeFileSync("./swagger-spec.json", JSON.stringify(document, null, 2));
```
Comment (JaJako) [inline backend/src/main.ts:17]: Good idea - I've already wondered if there's some solution not to have a one-liner in swagger-spec.json :)

File: frontend/src/api/models/LegInfo.ts
Comment (JaJako) [inline frontend/src/api/models/LegInfo.ts:1]: Why does the generate-client adds another final line break here? When I open this file on my machine or in GitHub's editor there are two final line breaks with this change now (not just the one that's intended). I've already stumbled over this a few weeks ago when I was looking in the CR/LF issue regarding client generation. There were some files checked in with two final line breaks which the client generation on my machine reduced to just one (causing another change after generation).

File: backend/src/journey/service/vrrJourneysWrapper.service.ts [hunk 2]
```diff
@@ -96,11 +99,21 @@ export class VrrJourneysWrapperService {
...
+    const realtimeStatusMap: Record<RealtimeTripStatus, LegRealtimeTripStatus> = {
+      DEVIATION: LegRealtimeTripStatus.deviation,
+      TRIP_CANCELLED: LegRealtimeTripStatus.tripCancelled,
+      EXTRA_STOPS: LegRealtimeTripStatus.extraStops,
+      EXTRA_TRIP: LegRealtimeTripStatus.extraTrip,
+      MONITORED: LegRealtimeTripStatus.monitored,
+      OUTSIDE_REALTIME_WINDOW: LegRealtimeTripStatus.outsideRealtimeWindow,
+      PROGNOSIS_IMPOSSIBLE: LegRealtimeTripStatus.prognosisImpossible,
+      REALTIME_ONLY_INFORMATIVE: LegRealtimeTripStatus.realtimeOnlyInformative
+    };
```
Comment (JaJako) [inline backend/src/journey/service/vrrJourneysWrapper.service.ts:111]: Needing to change this hard-coded mapping on changes to VRR's or our enum is not ideal, but I don´t have a better solution for this. Doing it this way there's no unnecessary complexity at least.

**Comment by JaJako** [comment]
There's another enum type on VRR's end that's dealt as a string on our end: `type` in location entity. 

But after looking at our current main branch's application I'm not sure if we are getting different location types any more - the journeyDemo is stating "unsupported" for every found location. Is this a intended behavior caused by the changed vrr parameters in query?

**Review by JaJako — CHANGES_REQUESTED** [review]
Looking good so far. If 'type' on location entity is still needed this should be changed to an enum, too. Maybe you could also take a look at the double line break in your client-generated files.

File: frontend/src/api/models/LegInfo.ts
Comment (JaJako) [reply frontend/src/api/models/LegInfo.ts:1]: Just saw your branch after I've created an issue for fixing that Dates are not converted from string to Date in backend. That's solved by the new generation, right? Maybe CR/LF handling gets improved, too :D

**Comment by Paulomart** [comment]
@JaJako Do you know how the `@Inject` stuff in nestjs works? Im done with my implementation but I just breaks because I dont know how to use the DI

**Comment by JaJako** [comment]
> @JaJako Do you know how the `@Inject` stuff in nestjs works? Im done with my implementation but I just breaks because I dont know how to use the DI

Spontaneously I have no idea, but I can take a look at the branch - maybe I see something.

**Comment by Paulomart** [comment]
@JaJako just solved it :D

**Comment by Paulomart** [comment]
@JaJako should be done now

**Review by JaJako — APPROVED** [review]

**Comment by Paulomart** [comment]
Hmm okay, the tests are failing for some technical import error now. aghr

**Comment by Paulomart** [comment]
Interesting, GitHub merged the pr even thou the tests were failing...

**Pull request merged**